src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -7841,4 +8010,271 @@ void wm_vuldel_truncate_revision(char * revision) {
     return;
 }
 
+int wm_vuldet_json_alas_parser(cJSON *json_feed, wm_vuldet_db *parsed_vulnerabilities) {
+    cJSON *advisories = NULL;
+    cJSON *vulnerability = NULL;
+    cJSON *vuln_node = NULL;
+    vu_alas_vuln *alas_vuln = NULL;
+
+    static char *JSON_ADVISORIES = "advisories";
+    static char *JSON_SEVERITY = "severity";
+    static char *JSON_PUBLISHED_DATE = "publishedDate";
+    static char *JSON_LAST_MODIFIED_DATE = "lastModifiedDate";
+    static char *JSON_REFERENCES = "references";
+    static char *JSON_VULNERABILITIES = "vulnerabilities";
+    static char *JSON_FIXED_PACKAGES = "fixed_packages";
+
+    if (advisories = cJSON_GetObjectItem(json_feed, JSON_ADVISORIES), !advisories) {
+        return OS_INVALID;
+    }
+
+    int i = 0;
+    for (advisories = advisories->child; json_tagged_obj(advisories); advisories = advisories->next) {
+        if (alas_vuln) {
+            os_calloc(1, sizeof(vu_alas_vuln), alas_vuln->next);
+            alas_vuln->next->prev = alas_vuln;
+            alas_vuln = alas_vuln->next;
+        } else {
+            os_calloc(1, sizeof(vu_alas_vuln), alas_vuln);
+            parsed_vulnerabilities->alas_vuln = alas_vuln;
+        }
+        w_strdup(advisories->string, alas_vuln->id);
+
+        for (vuln_node = advisories->child; json_tagged_obj(vuln_node); vuln_node = vuln_node->next) {
+
+            if (!vuln_node->string) {
```

Comment:
You were right. The check was duplicated so removing one of them. `valuestring` is not always non-null since there exist arrays where the content is taken from `child` instead of `valuestring`.